pkgs/applications/editors/emacs-modes/emacs-w3m/default.nix

```diff
@@ -1,19 +1,21 @@
-{ fetchcvs, stdenv, emacs, w3m, imagemagick, texinfo, autoreconfHook }:
+{ fetchFromGitHub, fetchgit, git, stdenv, emacs, w3m, imagemagick, texinfo, autoreconfHook }:
 
-let date = "2013-03-21"; in
+let date = "2019-02-20"; in
 stdenv.mkDerivation rec {
-  name = "emacs-w3m-cvs${date}";
-
-  # Get the source from CVS because the previous release (1.4.4) is old and
-  # doesn't work with GNU Emacs 23.
-  src = fetchcvs {
-    inherit date;
-    cvsRoot = ":pserver:anonymous@cvs.namazu.org:/storage/cvsroot";
-    module = "emacs-w3m";
-    sha256 = "1lmcj8rf83w13q8q68hh7sa1abc2m6j2zmfska92xdp7hslhdgc5";
+  name = "emacs-w3m-${date}";
+
+  # w3mhack.el expects the build dir to be a git clone repo. It
+  # looks into .git with the git executable and only does so if
+  # .git/config is present. Otherwise the build fails.
+
+  src = fetchgit {
+    url = "http://github.com/emacs-w3m/emacs-w3m";
+    rev = "7fa00f90b3bff8562768d0a15b32012cededcb47";
+    sha256 = "15q40aqfda40slkgs0fg0svm0x47a98nz81ysxb5nh0kr45ha67i";
+    leaveDotGit = true;
```

Comment:
This can easily break the checksum if git is updated.
Depending on how it uses git. The git binary can be also stubbed out:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/misc/flameshot/default.nix#L26

The melpa version would be also fine. Does it have w3m available automatically?